vision_datasets/common/manifest_dataset.py

```diff
@@ -172,6 +209,92 @@ def crop(img, left, t, r, b, aug_params=None, rnd: random.Random = None):
 
         return crop_img
 
+    def generate_manifest(self, **kwargs):
+        """
+        Generate dataset manifest for the multiclass classification dataset converted from detection dataset by cropping bboxes as classification samples.
+        Crops will be saved into 'dir' for generating the manifest
+        Args:
+            'dir'(str): directory where cropped images will be saved
+            'n_copies'(int): number of image copies generated for each bbox
+        """
+
+        local_cache_params = {'dir': kwargs.get('dir', f'{self.dataset_info.name}-cropped-ic'), 'n_copies': kwargs.get('n_copies')}
+        cache_decor = LocalFolderCacheDecorator(self, local_cache_params)
+        return cache_decor.generate_manifest()
+
+
+class LocalFolderCacheDecorator(BaseDataset):
+    """
+    Decorate a dataset by caching data in a local folder, in local_cache_params['dir'].
+
+    """
+
+    def __init__(self, dataset: BaseDataset, local_cache_params: dict):
+        """
+        Args:
+            dataset: dataset that requires cache
+            local_cache_params(dict): params controlling local cache for image access:
+                'dir': local dir for caching crops, it will be auto-created if not exist
+                [optional] 'n_copies': default being 1. if n_copies is greater than 1, then multiple copies will be cached and dataset will be n_copies times bigger
+        """
+
+        assert dataset is not None
+        assert local_cache_params
+        assert local_cache_params.get('dir')
+        local_cache_params['n_copies'] = local_cache_params.get('n_copies', 1)
+        assert local_cache_params['n_copies'] >= 1, 'n_copies must be equal or greater than 1.'
+
+        super().__init__(dataset.dataset_info)
+
+        self._dataset = dataset
+        self._local_cache_params = local_cache_params
+        if not os.path.exists(self._local_cache_params['dir']):
+            os.makedirs(self._local_cache_params['dir'])
```

Comment:
is this dir just a temporary cache, or a permanent directory for the dataset?

Reply:
It is user's choice

Reply:
got it